src/cocotb/task.py

```diff
@@ -157,6 +157,10 @@ def _advance(self, outcome: Outcome) -> Any:
         except StopIteration as e:
             self._outcome = Value(e.value)
             self._state = Task._State.FINISHED
+        except (KeyboardInterrupt, SystemExit):
+            # Allow these to bubble up to the execution root to fail the sim immediately.
+            # This follow's asyncio's behvaior.
```

Comment:
```suggestion
            # This follows asyncio's behvaior.
```

Reply:
I surprised you missed `behvaoir`! But I'll fix this.